spotify/models/user.py

```diff
@@ -144,22 +109,13 @@ def __init__(
         else:
             self.library = Library(client, self)
 
-        self.token_info: TokenInfo = token_info or TokenInfo()
+        self.token_info = token_info
 
         self._cache_path: Optional[Path] = kwargs.pop("cache_path", None)
-        if (
-            self._cache_path and not self._cache_path.exists()
-        ):  # make sure the file and folders exist
-            self._cache_path.parent.mkdir(parents=True, exist_ok=True)
-            self._cache_path.touch(exist_ok=True)
-        self._save_to_cache(self.token_info)
-
-        if refresh and self.token_info.refresh_token and self.token_info.expires_in:
-            coroutine: Coroutine = self._refreshing_token(
-                self.token_info.expires_in, self.token_info.refresh_token
-            )  # pylint: disable=protected-access
-
-            self._refresh_task = self.client.loop.create_task(coroutine)  #
+        if self._cache_path and not self._cache_path.exists():
+            raise FileNotFoundError("Cache file does not exist.")
+        elif self._cache_path and self.token_info:
+            self.token_info.save_to_file(self._cache_path)
```

Comment:
See my previous comment about expensive initialization, this is still potentially as expensive if not more expensive than previously.